pyEpiabm/pyEpiabm/tests/test_sweep/test_host_progression_sweep.py

```diff
@@ -73,6 +98,76 @@ def test_update_time(self):
         self.assertIsInstance(self.person1.time_of_status_change, float)
         self.assertTrue(6.0 <= self.person1.time_of_status_change <= 15.0)
 
+    def test_update_next_infection_status(self):
+        test_sweep = pe.sweep.HostProgressionSweep()
+        # Check assertion is raised if length of weights
+        # and outcomes are different
+        test_sweep.state_transition_matrix['Test col'] = ""
+        with self.assertRaises(AssertionError):
+            test_sweep._update_next_infection_status(self.people[0])
+
+        # Check that method works with identity state matrix
+        identity_matrix = pd.DataFrame(np.identity(len(InfectionStatus)),
+                                       columns=['Susceptible', 'Exposed',
+                                                'InfectASympt', 'InfectMild',
+                                                'InfectGP', 'InfectHosp',
+                                                'InfectICU', 'InfectICURecov',
+                                                'Recovered', 'Dead'],
```

Comment:
You use this list multiple times in the tests - probably easier to define it as a variable so you can pass that around instead (and only need to change one line if you ever add extra states)

EDIT: In fact, it might be even nicer to read in a list of states from the InfectionStatus enum instead?